core/src/test/java/org/opensearch/sql/expression/datetime/DateDiffTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.datetime;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.temporal.Temporal;
+import java.util.TimeZone;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class DateDiffTest extends DateTimeTestBase {
+
+  // Function signature is:
+  // (DATE/DATETIME/TIMESTAMP/TIME, DATE/DATETIME/TIMESTAMP/TIME) -> LONG
+  private static Stream<Arguments> getTestData() {
```

Comment:
Should add test cases covering the comments I made in the `conversations` tab.

Reply:
> When second argument is larger than the first argument the returned time is second argument
> 
> Should give negative time:
> 

Unfortunately, there is no negative time or time > 24 hours in OpenSearch.
If you start a 22 hour timer at 22:00, it will stop at 20:00 next day. Try playing with other numbers to avoid being confused, e.g. 7 AM/PM and 10 AM/PM.